File: chatbot.java
import java.util.Scanner;

public class chatbot {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Hello! I'm your friendly chatbot. What's your name?");
        String name = scanner.nextLine();
        System.out.println("Nice to meet you, " + name + "! How can I assist you today?");

        String input;
        do {
            System.out.print("> ");
            input = scanner.nextLine().toLowerCase();

            String response = getResponse(input);
            System.out.println(response);
        } while (!input.equals("bye"));

        System.out.println("Goodbye, " + name + "! Have a great day.");
        scanner.close();
    }

    public static String getResponse(String input) {
        switch (input) {
            case "hi":
            case "hello":
                return "Hi there! How can I help you?";
            case "how are you":
                return "I'm just a chatbot, but thanks for asking!";
            case "bye":
                return "Goodbye!";
            default:
                return "I'm not sure how to respond to that. Can you ask me something else?";
        }
    }
}

File: chatbot.py
import random
def greet():
    responses = ["Hi there! I'm your friendly chatbot.", 
                 "Hello! How can I assist you today?",
                 "Hey! Welcome to our chat. How can I help you?"]
    return random.choice(responses)
def ask_name():
    responses = ["What's your name?", 
                 "May I know your name?", 
                 "Could you please tell me your name?"]
    return random.choice(responses)
def handle_name(user_input):
    name = user_input.capitalize()
    return f"Nice to meet you, {name}!"
def respond(user_input):
    responses = ["I'm not sure I understand. Can you please clarify?", 
                 "Sorry, I didn't get that. Could you rephrase?", 
                 "Hmm, I didn't catch that. Can you say it differently?"]
    return random.choice(responses)
def chatbot():
    print(greet())
    print(ask_name())
    user_input = input("You: ")
    print(handle_name(user_input))
    
    while True:
        user_input = input("You: ").lower()
        
        if user_input == "exit":
            print("Goodbye! Have a great day.")
            break
        else:
            print(respond(user_input))
if __name__ == "__main__":
    chatbot()
